modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/edit_workflow_definition.jsp

```diff
@@ -27,7 +27,9 @@ portletDisplay.setURLBack(
 		"/view.jsp"
 	).buildString());
 
-renderResponse.setTitle((kaleoDefinitionVersion == null) ? LanguageUtil.get(request, "new-workflow-definition") : LanguageUtil.get(request, "edit-workflow-definition"));
+boolean isView = Objects.equals((String)request.getParameter(WorkflowWebKeys.WORKFLOW_JSP_STATE), "view");
+
+renderResponse.setTitle((kaleoDefinitionVersion == null) ? LanguageUtil.get(request, "new-workflow-definition") : LanguageUtil.get(request, (isView ? "view" : "edit") + "-workflow-definition"));
```

Comment:
Hi @cgmonte,

what do you think about this approach below?

````

String titleKey = "new-workflow-definition";

if (kaleoDefinitionVersion != null) {
  titleKey = "edit-workflow-definition";

  if (isView) {
     titleKey = "view-workflow-definition";
  }
}

renderResponse.setTitle(LanguageUtil.get(request, titleKey));

````

Reply:
Looks much cleaner indeed!

Reply:
I see you have edited. I prefer the previous because is more simples and explicit. Also, although this one is more concise, it does the same amount of if tests, so I left like you first suggested.